agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1642,7 +1642,13 @@ func TestFirelensWithS3ConfigFluentd(t *testing.T) {
 		Key:    aws.String("testfiles/fluentd.conf"),
 		Body:   strings.NewReader(content),
 	})
-	require.NoError(t, err, "unable to upload config file to s3 which is needed for the test")
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "AccessDenied" {
```

Comment:
having this as a string is fine now, but if we could have it reference an actual error such as `s3.ErrAccessDenied`, it would be better. 
also, this can be made a constant.

Reply:
i looked at aws go sdk, but it doesn't seem like s3 is throwing a specific error for the api, or even defining the error code as a constant anywhere. i updated the error code to be a constant